tmt/utils.py

```diff
@@ -299,6 +299,11 @@ def _save_context(cls, context: click.Context) -> None:
         cls._context = context
         cls._options = context.params
 
+    def _save_context_to_instance(self, context: click.Context) -> None:
```

Comment:
This was a necessary addition. There are 2 types of cases where we save the context:
 1) in CLI to the class (relevant to the command/subcommand)
 2) in Common's constructor. In this case, we want to override it in the instance.

The previous implementation caused a problem with clean where multiple instances of the `Clean` class are constructed at first and then executed at the end one after another. By overriding the class's context, only the latest context was kept, causing option inheritance to break.

There are solutions which could allow us to have one method and determine based on the binding, see https://stackoverflow.com/questions/28237955/same-name-for-classmethod-and-instancemethod but I think such approaches are more complicated and confusing than having 2 methods, one for binding the class and the other for binding the instance. @psss any thoughts on this?

Reply:
Thanks for the detailed explanation. Sounds good to me.